docs/upgrading-openshift.md

```diff
@@ -60,6 +62,130 @@ passwd:
+boot_device:
+  luks:
+    tang:
+      - url: https://tang.example.com
+        thumbprint: REPLACE-THIS-WITH-YOUR-TANG-THUMBPRINT
+        advertisement: "{\"payload\": \"...\", \"protected\": \"...\", \"signature\": \"...\"}"
+storage:
+  luks:
+    - name: luks-tang
+      device: /dev/sdb
+      clevis:
+        tang:
+          - url: https://tang.example.com
+            thumbprint: REPLACE-THIS-WITH-YOUR-TANG-THUMBPRINT
+            advertisement: "{\"payload\": \"...\", \"protected\": \"...\", \"signature\": \"...\"}"
+```
+
+### LUKS discard
+
+The `luks` sections in `storage` and `boot_device` gained a new `discard` field. If specified and true, the LUKS volume will issue discard commands to the underlying block device when blocks are freed. This improves performance and device longevity on SSDs and space utilization on thinly provisioned SAN devices, but leaks information about which disk blocks contain data.
+
+<!-- butane-config -->
+```yaml
+variant: openshift
+version: 4.14.0
+metadata:
+  name: minimal-config
+  labels:
+    machineconfiguration.openshift.io/role: worker
+boot_device:
+  luks:
+    discard: true
+    tpm2: true
+storage:
+  luks:
+    - name: luks-tpm
+      device: /dev/sdb
+      discard: true
+      clevis:
+        tpm2: true
+```
+
+### LUKS open options
+
+The `storage.luks` section gained a new `open_options` field. It is a list of options Ignition should pass to `cryptsetup luksOpen` when unlocking the volume. Ignition also passes `--persistent`, so any options that support persistence will be saved to the volume and automatically used for future unlocks. Any options that do not support persistence will only be applied to Ignition's initial unlock of the volume.
+
+<!-- butane-config -->
+```yaml
+variant: openshift
+version: 4.14.0
+metadata:
+  name: minimal-config
+  labels:
+    machineconfiguration.openshift.io/role: worker
+storage:
+  luks:
+    - name: luks-tpm
+      device: /dev/sdb
+      open_options:
+        - "--perf-no_read_workqueue"
+        - "--perf-no_write_workqueue"
+      clevis:
+        tpm2: true
+```
+
+### Local SSH key and systemd unit references
+
+SSH keys and systemd units are now embeddable via file references to local files. The specified path is relative to a local _files-dir_, specified with the `-d`/`--files-dir` option to Butane. If no _files-dir_ is specified, this functionality is unavailable.
+
+<!-- butane-config -->
+```yaml
+variant: openshift
+version: 4.14.0
+metadata:
+  name: minimal-config
+  labels:
+    machineconfiguration.openshift.io/role: worker
+systemd:
+  units:
+    - name: example.service
+      contents_local: example.service
+    - name: example-drop-in.service
+      dropins:
+        - name: example-drop-in.conf
+          contents_local: example.conf
+passwd:
+  users:
+    - name: core
+      ssh_authorized_keys_local:
+        - id_rsa.pub
+```
+
+### Automatic generation of systemd swap units
```

Comment:
I wonder if we support swap in OCP. Maybe we do now but we did not in the past. Let's check that.

Reply:
This will not work, good catch; no support for filesystem changes after day-0

Reply:
This falls under the general case of https://issues.redhat.com/browse/MCO-1002 I believe, where right now it's only allowed at install time

Reply:
So to be clear, since this is talking about setting swap at install time this correct to document? @yuqi-zhang

Reply:
Yes, I was unsure as well because the change is under `docs/upgrading-openshift.md`

If this is install time only, let's make sure we clarify that.